packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -323,7 +334,10 @@ export class LoadBalancer extends Resource implements IConnectable {
 
   public addTarget(target: ILoadBalancerTarget) {
     target.attachToClassicLB(this);
-
+    if (target.ec2Instance) {
+      this.instanceIds.push(target.ec2Instance.instanceId);
+      target.ec2Instance.addSecurityGroup(this.securityGroup);
```

Comment:
I don't think we need to do `addSecurityGroup`.

We need to do something like:

```ts
target.ec2Instance.connections.allowFrom(/* something here */);
```

And it would be better if that implementation was inside `attachToClassicLB`, because that is ultimately the function that should do the work of "connecting an LB to an instance".